packages/access-client/test/agent-use-cases.test.js

```diff
@@ -188,3 +192,50 @@ describe('authorizeWaitAndClaim', async function () {
     assert(claimHandler.notCalled)
   })
 })
+
+describe('getAccountPlan', async function () {
+  const accountWithAPlan = 'did:mailto:example.com:i-have-a-plan'
+  const accountWithoutAPlan = 'did:mailto:example.com:i-have-no-plan'
+
+  /** @type {Record<Ucanto.DID, {product: Ucanto.DID, updatedAt: string}>} */
+  const plans = {
+    [accountWithAPlan]: {
+      product: 'did:web:test.web3.storage',
+      updatedAt: new Date().toISOString()
+    }
+  }
+
+  const server = createServer({
+    plan: {
+      get: Server.provide(Plan.get, ({ capability }) => {
+        const plan = plans[capability.with]
+        if (plan) {
+          return { ok: plan }
+        } else {
+          return {
+            error: {
+              name: 'PlanNotFound',
+              message: ''
+            }
+          }
+        }
+      }),
+    },
+  })
+  const agent = await Agent.create(undefined, {
+    connection: connection({ principal: server.id, channel: server }),
+  })
+
+  await Promise.all([
+    ...await createAuthorization({ account: accountWithAPlan, agent: agent.issuer, service: server.id }),
+    ...await createAuthorization({ account: accountWithoutAPlan, agent: agent.issuer, service: server.id })
+  ].map(proof => agent.addProof(proof)))
+
+  it("should succeed for accounts with plans", async function () {
+    assert((await getAccountPlan(agent, accountWithAPlan)).ok)
+  })
+
+  it("should fail for accounts without a plan", async function () {
```

Comment:
could be nice to assert the error here. i.e. if I am a client and I try to plan/get, and there is an error, how do I know whether the error is a sign that my account has no plan (but a UI could still help user set one) vs a sign of something more wrong in the system (and e.g. a UI should show a 'come back later' state if the error is unknown).

Reply:
e.g. I think this would be a PlanNotFound ?

Reply:
great idea - done